### ISSUE TITLE
Check modules and methods regarding their name and location

### ISSUE THREAD

**Issue by ST-DDT** [faker-js/faker#1346]
### Clear and concise description of the problem

Some methods are misplaced or misnamed. The same is true for some module.

E.g. the string and number methods, or the address and name modules

### Suggested solution

Check every module and method regarding their current name/location.

**Comment by ST-DDT** [comment]
See also
- #805

**Comment by ST-DDT** [comment]
Blocked by
- #1341
- #1342 
- #1343 
- #1344 
- #1350

**Comment by xDivisionByZerox** [comment]
This should no longer be blocked. Correct, @ST-DDT?

**Comment by ST-DDT** [comment]
@xDivisionByZerox Yeah, you are right. Do you wish to tackle this?

**Comment by xDivisionByZerox** [comment]
Checklist from my side:

| Module         | State |
| -------------- | ----- |
| Animal   | ✅    |
| Color    | ✅    |
| Commerce | ✅    |
| Company| <ul><li> `bs` should probably be deprecated #1860</li><li>`bsAdjective` should probably be merged into `catchPhraseAdjective` and deprecate #1860</li><li>`bsBuzz` should probably be called `buzzWord` (is that correctly written?) #1860</li><li>`bsNoun` should probably be merged into `catchPhraseNoun` and deprecated #1860</li></ul>   |
|Database|✅|
|Datatype|#1590|
|Date|<ul><li>I don't really know why we have `between` AND `betweens`</li><li>potentially `month` should be called `monthName`</li></ul>|
|Finance|<ul><li>`account` should probably be called `accountNumber` #2054</li><li>❌ `iban` should perhaps be called `IBAN`</li><li>`mask` should be renamed to `maskedNumber` or be removed #2055</li></ul>|
|Git|<ul><li>✅ `shortSha` should probably be merged with `commitSha` and configurable via an argument</li></ul>|
|Hacker|<ul><li>`ingverb` should probably be called `ingVerb` (or generally a better name)</li></ul>|
|Helpers|✅|
|Image|✅|
|Internet|<ul><li>There is `email` AND `exampleEmail` (what is an example email?)</li></ul>|
|Location|<ul><li>`city` and `cityName` should probably be merged</li><li>`nearbyGPSCoordinate` perhaps could be redefined to `coordinates`</li><li>`state` and `stateAbbr` should probably be merged #2060</li><li>`streetAddress`</li><li>Having `street`, `streetAddress`, `streetName` AND `secondaryAddress` is very confusing to me</li><li>`street` and `streetName` should probably be merged #2051</li></ul>|
|Lorem|<ul><li>`slug` seems out of place for me here</li><li>There is `paragraphs` and `text` - aint that the same thing?</li></ul>|
|Music |✅|
|Number|Maybe a bit weird that `int` and `hex` use abbreviation while `binary` and `octal` don't, but ✅|
|Person|<ul><li>`job*` could fit in the `CompanyModule` as well</li></ul>|
|Phone|✅|
|Random|<ul><li>Having a `word` function while we have an entire `WordModule` seems off (same for `words`)</li><li>`locale` should probably be located in the `LocationModule`</li><li>The `RandomModule` could be dropped entirely</li></ul>|
|Science|✅|
|String|<ul><li>`fromCharacters` should probably be located in the `HelpersModule`</li><li>✅ `special` should probably be called `symbol` #1711</li></ul>|
|System|✅|
|Vehicle|<ul><li>`vin` should probably be called `identificationNumber` for readability purposes</li><li>`vrm` should probably be called `registrationMark` for readability purposes</li></ul>|
|Word|✅|

**Comment by ejcheng** [comment]
> Checklist from my side:
> 
> Module	State
> Animal	✅
> Color	✅
> Commerce	✅
> Company	
> * `bs` should probably be deprecated
> * `bsAdjective` should probably be merged into `catchPhraseAdjective` and deprecate
> * `bsBuzz` should probably be called `buzzWord` (is that correctly written?)
> * `bsNoun` should probably be merged into `catchPhraseNoun` and deprecated
> 
> Database	✅
> Datatype	#1590
> Date	
> * I don't really know why we have `between` AND `betweens`
> * potentially `month` should be called `monthName`
> 
> Finance	
> * `account` should probably be called `accountNumber`
> * `iban` should perhaps be called `IBAN`
> * `mask` should be renamed to `maskedNumber` or be removed
> 
> Git	
> * `shortSha` should probably be merged with `commitSha` and configurable via an argument
> 
> Hacker	
> * `ingverb` should probably be called `ingVerb` (or generally a better name)
> 
> Helpers	✅
> Image	✅
> Internet	
> * There is `email` AND `exampleEmail` (what is an example email?)
> 
> Location	
> * `city` and `cityName` should probably be merged
> * `nearbyGPSCoordinate` perhaps could be redefined to `coordinates`
> * `state` and `stateAbbr` should probably be merged
> * `streetAddress`
> * Having `street`, `streetAddress`, `streetName` AND `secondaryAddress` is very confusing to me
> * `street` and `streetName` should probably be merged
> 
> Lorem	
> * `slug` seems out of place for me here
> * There is `paragraphs` and `text` - aint that the same thing?
> 
> Music	✅
> Number	Maybe a bit weird that `int` and `hex` use abbreviation while `binary` and `octal` don't, but ✅
> Person	
> * `job*` could fit in the `CompanyModule` as well
> 
> Phone	✅
> Random	
> * Having a `word` function while we have an entire `WordModule` seems off (same for `words`)
> * `locale` should probably be located in the `LocationModule`
> * The `RandomModule` could be dropped entirely
> 
> Science	✅
> String	
> * `fromCharacters` should probably be located in the `HelpersModule`
> * `special` should probably be called `symbol` [Generate random symbol (eg: ! _ #) #1711](https://github.com/faker-js/faker/issues/1711)
> 
> System	✅
> Vehicle	
> * `vin` should probably be called `identificationNumber` for readability purposes
> * `vrm` should probably be called `registrationMark` for readability purposes
> 
> Word	✅

About `lorem.slug`, perhaps we could move it to a new method `internet.slug`? I feel like it would make more sense in `internet`. I assume we would have to deprecate `lorem.slug` first though

**Comment by xDivisionByZerox** [comment]
✅ **Team Decision: Company**

> `bs` should probably be deprecated

Yes!

> `bsAdjective` should probably be merged into `catchPhraseAdjective` and deprecate
> `bsBuzz` should probably be called `buzzWord` (is that correctly written?)
> `bsNoun` should probably be merged into `catchPhraseNoun` and deprecated

Replace `bs*` with `buzz*` and improve JSDocs (buzz = manager language, catchPhrase = user language).

**Comment by xDivisionByZerox** [comment]
**Team Decision: Date**

> I don't really know why we have `between` AND `betweens`

To be decided later.

> potentially `month` should be called `monthName`

Sounds good. More information will be discussed in an individual issue.

**Comment by xDivisionByZerox** [comment]
✅ **Team Decision: Finance**

> `account` should probably be called `accountNumber`

Sounds good.

> `mask` should be renamed to `maskedNumber` or be removed

Renaming sounds good.

> `iban` should perhaps be called `IBAN`

Nope, this is naming convention. Will not be changed.

**Comment by xDivisionByZerox** [comment]
✅ **Team Decision Git** 

> `shortSha` should probably be merged with `commitSha` and configurable via an argument

Sound good. Can be a `length` param.

**Comment by xDivisionByZerox** [comment]
**Team Decision: Hacker**

> `ingverb` should probably be called `ingVerb` (or generally a better name)

We want to give this a more descriptive name but will discuss the details later. We can also consider removing but require more information for other languages.

**Comment by xDivisionByZerox** [comment]
**Team Decision: Internet**

> There is `email` AND `exampleEmail` (what is an example email?)

[rfc2606 - .example domain](https://www.rfc-editor.org/rfc/rfc2606)
We can merge both via an option flag in `email` and `url`.

**Comment by xDivisionByZerox** [comment]
**Team Decision: Location**

> `state` and `stateAbbr` should probably be merged

Sounds good.

> `city` and `cityName` should probably be merged
> `street` and `streetName` should probably be merged
> `nearbyGPSCoordinate` perhaps could be redefined to `coordinates`

Discuss all naming later.

> Having `street`, `streetAddress`, `streetName` AND `secondaryAddress` is very confusing to me

Discuss later.

---

✅ We also want to merge `zipCode` and `zipCodeByState`. `zipCode` should then throw an error if the provided `state` argument is unknown.

**Comment by Shinigami92** [comment]
It seems we have tackled all needed changes. If we want to change something additionally, we should open issues/PRs specifically.
